### ISSUE TITLE
Adding the menu style and functionality

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#67]
The menu style needs to be added and the functionality must be carefully tested for any accessibility problems.

Design: 
https://www.figma.com/file/OEtsCj8oD8JPVx4OvsS0U2/Twenty-Twenty-One?node-id=487%3A61

- [x] The toggled menu should be closable with the Esc key. 

- [x] The modal only closes when the user actively closes it.

- [x] -When opened, the focus is kept inside the menu modal.  Including the site title or logo in the focus order is acceptable (when it is linked) 

-The opened responsive menu will cover the full viewport.

-The existing markup for the toggle and close button should be simplified if possible.

-All controls and menu items must work/be able to be accessed when using the keyboard. 

-The sub menu items should also be navigable with arrow keys. 

-Ideally the user should be able to chose if they want to tab through all the sub menu items, or go directly to the next parent menu item. Avoid unnecessary tabbing.

**Block styles**
By assuring that the primary menu and the **navigation block** (both variations) looks the same,
a user can easily add secondary menus that matches the design and that are accessible.

The menu needs to be tested both with and without Gutenberg installed because there are some known differences in how the colors are used.

**Comment by melchoyce** [comment]
>  Responsive sub menu items: show or hide, decision needed.

We've decided to show submenu items by default on small screens.

**Comment by melchoyce** [comment]
> Responsive sub menu items: show or hide, decision needed.
> The toggled menu and sub menu items should be closable with the Esc key.
> The modal only closes when the user actively closes it.

Addressed in #115.

> Add fallback page menu -decision needed

Can I get some more clarification around this? I'm not sure what it means.

**Comment by pattonwebz** [comment]
> Can I get some more clarification around this? I'm not sure what it means.

When no menu exists in a given location the default is to use a fallback which is the the `wp_page_menu()`. If we don't want that then we need to explicitly disable it. https://developer.wordpress.org/reference/functions/wp_nav_menu/

**Comment by melchoyce** [comment]
Yeah, if there isn't a menu assigned to a location, I don't think we should show anything.

**Comment by pattonwebz** [comment]
> Yeah, if there isn't a menu assigned to a location, I don't think we should show anything.

I agree :) We can set `'fallback_cb' => false` to do that

**Comment by carolinan** [comment]
There is already a "has nav menu" wrapping the primary menu, but the above code is shorter.

**Comment by LittleBigThing** [comment]
A small suggested tweak for the primary navigation menu: #170
